RiotSwiftUI/Modules/Common/Util/MultilineTextField.swift

```diff
@@ -0,0 +1,213 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+@available(iOS 14.0, *)
+struct MultilineTextField: View {
```

Comment:
Is there a reason you didn't use [TextEditor](https://developer.apple.com/documentation/swiftui/texteditor)?

Reply:
Yes, there's a couple of reasons: I needed something that supports focus rings, placeholders and which doesn't allow new lines. Also, I tried out the TextEditor and it made the whole form jump around when editing text.